pkg/config/generic_map.go

```diff
@@ -29,3 +29,12 @@ func (m GenericMap) Copy() GenericMap {
 
 	return result
 }
+
+func (m GenericMap) IsDuplicate() bool {
+	if duplicate, hasKey := m["Duplicate"]; hasKey {
+		if isDuplicate, ok := duplicate.(bool); ok && isDuplicate {
```

Comment:
We have `ConvertToBool()` that can handle cases where the type of `m["Duplicate"]` is not bool but is convertible to bool.
https://github.com/netobserv/flowlogs-pipeline/blob/3f2481ef3474ba0b07b595a02b50f813c9938916/pkg/pipeline/utils/convert.go#L220

I just noticed that `ConvertToBool()` is missing the trivial `case bool:`. If you think it's worth using `ConvertToBool()` here, the fix of adding the trivial case could be done in this PR as well.

Reply:
I'm getting some conflicts when I try to import `pipeline/utils`:
```
$ make build
/home/julien/dev/flowlogs-pipeline/bin/golangci-lint-v1.50.1 run --enable goimports --enable gofmt --enable ineffassign --timeout 5m
ERRO Running error: context loading failed: failed to load packages: failed to load with go/packages: internal error: go list gives conflicting information for package github.com/netobserv/flowlogs-pipeline/pkg/config [github.com/netobserv/flowlogs-pipeline/pkg/config.test] 
make: *** [Makefile:70: lint] Error 3
```

Any idea ? Should we move `utils` outside of the `pipeline` package ?

Reply:
I think this errors implies a cyclic reference: `config` imports `pipeline/utils` and `pipeline/utils` imports `config`.
Looking at `pipeline/utils` files, it seems that only `batcher.go` has an import to `config`.
https://github.com/netobserv/flowlogs-pipeline/blob/3f2481ef3474ba0b07b595a02b50f813c9938916/pkg/pipeline/utils/batcher.go#L6

Hm... I'm not sure what would be the most proper solution in this case.

Reply:
Yeah so we should either move away `Convert*` functions or `batcher.go` from `pipeline/utils`...
Or keep as is.

Reply:
I think I would go with the first option of moving `convert.go` to an "FLP-independent" utils package.

Reply:
https://github.com/netobserv/flowlogs-pipeline/pull/387/commits/70514bf5353307c03d92be333bf72a7299f369d8 here we go